### PR TITLE
adding profile dropdown

### PR DESCRIPTION
**Pull request by cxjohn**
https://github.com/user-attachments/assets/5da8c849-0906-4ca9-9fd7-685d074b1d03

### CODE REVIEW

File: lib/components/ProfileDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+
+  const handleClick = () => {
+    setIsClicked(!isClicked);
+    setIsOpen(!isClicked);
+  };
+
+  const buttonStyle =
```
Comment (joelburget) [inline lib/components/ProfileDropdown.tsx:34]: Worth making a `Button` component?

File: lib/components/ProfileDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+"use client";
+
+import { useState, useRef, useEffect } from "react";
+import Link from "next/link";
+import { logout } from "@/app/login/actions";
+
+export default function ProfileDropdown() {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isClicked, setIsClicked] = useState(false);
```
Comment (joelburget) [inline lib/components/ProfileDropdown.tsx:9]: The purpose of `isClicked` isn't clear -- add a comment?

File: lib/components/ProfileDropdown.tsx (hunk 1 above)
Comment (cxjohn) [reply lib/components/ProfileDropdown.tsx:34]: I don't think it makes sense at the moment, given one element is a Next.js Link component and the other is a regular button.

**Review by joelburget — APPROVED** [review]
Great 🥇

**Pull request merged**